### DISCUSSION TITLE
NVDA should handle nested landmarks

### DISCUSSION THREAD

**Discussion by alexstine** [nvaccess/nvda#15561] [Ideas]
### Is your feature request related to a problem? Please describe.

Currently, if HTML5 tags are used inside other tags, there is no easy way with NVDA to exit the inner most landmark with landmark navigation. Picture this.

```html
<main>
<h1>Article Title</h1>
<nav aria-label="Table of Contents">
<!-- Table of contents links -->
</nav>
<p>This is some article content.</p>
<h2>A Sub Heading</h2>
</main>
<footer>
<p>Copyright 2023 Example Corp</p>
</footer>
```

If you jump by landmarks by pressing D, you will enter the main landmark and then the navigation landmark for the table of contents, but there is no way to move passed this landmark as you would land in the content info landmark vs. being able to continue in main.

### Describe the solution you'd like

Should D move to the main landmark content if you are inside an inner landmark within main? Should Shift+D move to the main landmark, navigating backwards out of navigation?

### Describe alternatives you've considered

Dedicated shortcut to exit inner landmarks similar to the shortcut to skip passed tables?

### Additional context

Breadcrumbs and table of contents are the worse offenders currently.

**Comment by seanbudd** [comment]
As it's unclear what the intended behaviour should be here, we are transferring this is to a discussion

**Comment by Adriani90** [comment]
Yeah I think the most suitable solution here would be to handle every landmark as a container and using comma or shift+komma would pass the current container forwards or backwards.

I am not in favor of using d or shift+d multiple times for the same landmark. this might cause confusions.

so the expected behavior in in my view would be as follows:

Scenario I:
1. Press d: you land on the main landmark
2. Press d again, you land on the nav landmark
3. press d again, you land on footer
4. Press d again: no further landmarks

Scenario II:
1. Press d: you land on the main landmark
2. Press Comma or shift+comma: you exid the main landmark
3. Press comma or shift+comma: not in a container

Scenario III:
1. Press d: you land on main landmark
2. Press d again: you land on nav landmark
3. Press comma or shift+comma: you exit the nav landmark
4. Press comma or shift+comma again: you exit the main landmark
5. Press comma or shift+comma: not in a container

cc: @CyrilleB79 do you have any further thoughts on this?

This is indeed a valid issue and should be converted back into an  issue in my view.
